scrapyd_k8s/object_storage/libcloud_driver.py

```diff
@@ -0,0 +1,79 @@
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+from libcloud.storage.types import ObjectError, ContainerDoesNotExistError, ObjectDoesNotExistError, InvalidContainerNameError
+from libcloud.storage.providers import get_driver
+
+class LibcloudObjectStorage:
+
+    def __init__(self, config):
+        self._storage_provider = config.joblogs().get('storage_provider')
+        if self._storage_provider is None:
+            logger.error('Storage provider is not defined in the configuration.')
+            raise ValueError('Storage provider is not defined')
+
+        self._container_name = config.joblogs().get('container_name')
+        if self._container_name is None:
+            logger.error('Container name is not set in the configuration.')
+            raise ValueError('Container name is not set')
+
+        args_envs = config.joblogs_provider_args(self._storage_provider)
+        args = {}
+        for arg, env in args_envs.items():
+            env_value = os.getenv(env)
+            if env_value is None:
+                logger.error(f"Environment variable '{env}' for argument '{arg}' is not set.")
+                raise ValueError(f"Environment variable '{env}' for argument '{arg}' is not set.")
+            args[arg] = env_value
+
+        driver_class = get_driver(self._storage_provider)
+        print("CLOUD CREDENTIALS")
+        print(args)
+        try:
+            self.driver = driver_class(**args)
```

Comment:
Really happy that you've been able to generalize the storage driver configuration, not needing to hardcode the possible keys for storage providers anymore :)